### PR TITLE
Test: Added new tests for the e2e

### PR DESCRIPTION
**Pull request by aagusuab**
### Description of your changes

Fixes #

I have:
- Added two tests to check controller's validity on duplicate manifests on work objects.
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: tests/e2e/apply_test.go [hunk 1]
```diff
@@ -42,6 +43,19 @@ var _ = Describe("Work creation", func() {
...
+
+	WorkWithDuplicateManifestsContext(
+		"with a duplicate manifest: Configmap",
+		[]string{
+			"manifests/test-configmap.ns.yaml",
+			"manifests/test-configmap.ns.yaml",
+		})
+
+	MultipleWorkWithSameManifestContext(
+		"with two resource that has the same manifest: Deployment",
```
Comment (Ealianis) [inline tests/e2e/apply_test.go:55]: "with two manifests for the same resource"

File: tests/e2e/apply_test.go [hunk 2]
```diff
@@ -408,6 +422,165 @@ var WorkDeletedContext = func(description string, manifestFiles []string) bool {
...
+			)
+
+		})
+
+		AfterEach(func() {
+			err = deleteWorkResource(work.Namespace, work.Name)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("one of two manifests should be applied", func() {
```
Comment (Ealianis) [inline tests/e2e/apply_test.go:447]: I believe it should be  ```It("Should result in only one unique resource```
The By statements yield how that uniqueness is determined

File: tests/e2e/apply_test.go [hunk 3]
```diff
@@ -408,6 +422,165 @@ var WorkDeletedContext = func(description string, manifestFiles []string) bool {
...
+				if err != nil {
+					return false
+				}
+
+				appliedWorkTwo, err := spokeWorkClient.MulticlusterV1alpha1().AppliedWorks().Get(context.Background(), workTwo.Name, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+
+				return len(appliedWorkOne.Status.AppliedResources)+len(appliedWorkTwo.Status.AppliedResources) == 0
```
Comment (Ealianis) [inline tests/e2e/apply_test.go:552]: Why is this == 0 ? Should it not be 1?

**Review by Ealianis — APPROVED** [review]
LGTM w/ comments.

File: tests/e2e/apply_test.go (hunk 1 above)
Comment (aagusuab) [reply tests/e2e/apply_test.go]: Fixed.

File: tests/e2e/apply_test.go (hunk 2 above)
Comment (aagusuab) [reply tests/e2e/apply_test.go]: Fixed.

File: tests/e2e/apply_test.go (hunk 3 above)
Comment (aagusuab) [reply tests/e2e/apply_test.go]: Mistake, Fixed.

**Pull request merged**